packages/core/src/Core.ts

```diff
@@ -67,13 +67,22 @@ export class Core<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T>
             const previousSibling = range.start.previousSibling();
             if (previousSibling) {
                 previousSibling.removeBackward();
-            } else {
-                // Otherwise delete nodes between range and previous leaf.
-                const previousLeaf = range.start.previousLeaf(node => {
-                    return node !== range.start.parent;
-                });
-                if (previousLeaf) {
-                    range.setStart(previousLeaf, RelativePosition.AFTER);
+            } else if (range.startContainer.breakable) {
+                // Otherwise set range start at previous valid leaf.
+                let previous = range.startContainer.previous();
+                while (previous && !isLeaf(previous)) {
+                    const sibling = previous.previousSibling();
+                    if (sibling) {
+                        previous = sibling.lastLeaf();
+                    } else if (previous.parent?.breakable) {
+                        previous = previous.parent;
+                    } else {
+                        // Stop browsing at the first child of an unbreakable.
+                        previous = undefined;
```

Comment:
I don't like to use `return` in weird places because it makes the flow of the code hard to grasp. Though I'll have a second look to this function to see if I can improve it.

Reply:
I could not find a better way of doing it. And using `return` doesn't cut it for me. This `undefined` means something, it symbolizes the fact that the "previous" of an unbreakable, in the context of this command, is `undefined`. It's what an hypothetical `previousButDontBreakUnbreakables` would return and for me the while *is* that hypothetical function. The fact that we end up not doing anything later on if there is no `previous` is outside of the scope of the `while`.

In other words, the responsibility of the `while` block here is to compute the value of `previous` in this context. It turns out that the computed value of `previous` in this context is `undefined`. I don't want the while to take control of the entire command and decide that the command will stop there just because it was unable to compute a meaningful value for `previous`, it is outside of the scope I designed it for.